bcipy/tasks/main_frame.py

```diff
@@ -228,15 +238,39 @@ def decide_state_update(self):
         self.list_epoch[-1]['decision'] = decision
         return decision
 
-    def prepare_stimuli(self):
+    # TODO: think about where timing, color, is_txt, num_sti should go.
+    # e.g. is_txt is predefined with the alphabet directly
+    def _prepare_stimuli(self, timing=[1, 0.2], color=['red', 'white'],
+                         num_sti=1, is_txt=True):
         """ Given the alphabet, under a rule, prepares a stimuli for
             the next sequence
             Return:
                 stimuli(tuple[list[char],list[float],list[str]]): tuple of
                     stimuli information. [0]: letter, [1]: timing, [2]: color
                 """
-        stimuli = \
-            best_case_rsvp_seq_gen(self.alphabet, self.list_epoch[-1][
-                'list_distribution'][-1], num_sti=1, is_txt=self.is_txt_sti,
-                timing=self.stimuli_timing)
+
+        tmp = self.list_epoch[-1]['list_distribution'][-1][:]
+        stimuli_pos = self.query_method.update_query(tmp[:])
+
+        samples, times, colors = [], [], []
+        for idx_num in range(num_sti):
+            # append a fixation cross. if not text, append path to
+            # image fixation
+            if is_txt:
+                sample = ['+']
+            else:
+                sample = ['bcipy/static/images/bci_main_images/PLUS.png']
```

Comment:
This is another pattern that we have throughout the code base that would be good to have consolidated in a single place. It would be good to add a helper function (in bci_task_related.py?) with something like:

        def with_fixation(sample: List, is_text: bool = True) -> List:
            """Return a copy of sample with fixation item prepended."""
            fixation = '+' if is_txt else 'bcipy/static/images/bci_main_images/PLUS.png'
            return [fixation] + sample